src/edu/hillel/lesson19/variant1/FileLogger.java

```diff
@@ -0,0 +1,61 @@
+package edu.hillel.lesson19.variant1;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class FileLogger implements Logger {
+    private final FileLoggerConfiguration loggerConfiguration;
+    private File file;
+
+    public FileLogger() {
+        FileLoggerConfigurationLoader loggerConfigurationLoader = new FileLoggerConfigurationLoader();
+        loggerConfiguration = loggerConfigurationLoader.load("lesson19_properties.txt");
+    }
+
+    @Override
+    public void debug(String message) {
+        if (LoggingLevel.DEBUG.equals(loggerConfiguration.getLoggingLevel())) {
+            logging(String.format(loggerConfiguration.getFormat(), LocalDateTime.now(), LoggingLevel.DEBUG, message));
+        }
+    }
+
+    @Override
+    public void info(String message) {
+        logging(String.format(loggerConfiguration.getFormat(), LocalDateTime.now(), LoggingLevel.INFO, message));
+    }
+
+    private File getNewFile() {
+        String fileName;
+        File file;
+        int i = 0;
+        try {
+            Files.createDirectories(Paths.get(loggerConfiguration.getLogFileName()));
+            do {
+                i++;
+                fileName = loggerConfiguration.getLogFileName() + "/log_" + LocalDateTime.now().format(DateTimeFormatter.ofPattern("dd.MM.yyyy-HH:mm:ss")) + "." + i + ".txt";
+                file = new File(fileName);
```

Comment:
A bit confused of this construction. Why did you use cycle to create a file?

Reply:
According to task - file name contain seconds. But if we set for example cycle to 10000 b sleep for 1 ms - we get 10 file per second. So I decided to add iteration point to file name.
And we will get .1, .2 and so files

Reply:
I knowingly have done it. I As I think task incorrect - we can get mo than file limit in one second.
So I decided to handle this situation